modules/auth/ldap/ldap.go

```diff
@@ -353,20 +427,14 @@ func (ls *Source) SearchEntries() ([]*SearchResult, error) {
 		return nil, err
 	}
 
-	result := make([]*SearchResult, len(sr.Entries))
+	results := []*SearchResult{}
```

Comment:
Depending on how much filtering is done it's almost always better to preallocate the array.

Here I'm assuming that all or at least most of the entries will end up in the results.

You could preallocate half or some other amount.

```suggestion
	results := make([]*SearchResult, 0, len(sr.Entries))
```

Reply:
I think we can assume that most entries end up in the results because they are already filtered with the `userFilter` in https://github.com/stertingen/gitea/blob/ldap_groups/modules/auth/ldap/ldap.go#L416.
`ProcessUserEntries()` will return `nil` only in a few cases:
* `sanitizedGroupQuery()` fails (unlikely)
* LDAP search fails (unlikely)
* User is not in the group required for membership (more likely, but only if a group membership is required)
Also, if the LDAP groups feature is not used, `nil` will never be returned.